addons/binding/org.openhab.binding.smhi/src/main/java/org/openhab/binding/smhi/internal/SmhiHandler.java

```diff
@@ -0,0 +1,440 @@
+                return new DecimalType(
+                        hashParameters.get(CHANNEL_PRECIPITATION_CATEGORY_ID_JSON).getValues()[0].intValue());
+            case CHANNEL_MEAN_PRECIPITATION:
+                return new DecimalType(hashParameters.get(CHANNEL_MEAN_PRECIPITATION_JSON).getValues()[0]);
+            case CHANNEL_MEDIAN_PRECIPITATION:
+                return new DecimalType(hashParameters.get(CHANNEL_MEDIAN_PRECIPITATION_JSON).getValues()[0]);
+            case CHANNEL_MIN_PRECIPITATION:
+                return new DecimalType(hashParameters.get(CHANNEL_MIN_PRECIPITATION_JSON).getValues()[0]);
+            case CHANNEL_MAX_PRECIPITATION:
+                return new DecimalType(hashParameters.get(CHANNEL_MAX_PRECIPITATION_JSON).getValues()[0]);
+            case CHANNEL_FROZEN_PRECIPITATION:
+                double fp = hashParameters.get(CHANNEL_FROZEN_PRECIPITATION_JSON).getValues()[0];
+                if (fp == -9) {
+                    fp = 0;
+                }
+                return new DecimalType(fp);
+        }
+
+        return UnDefType.NULL;
+    }
+
+    public static boolean coordinate_is_inside_polygon(double latitude, double longitude,
+            ArrayList<String> polygon_lat_long_pairs) {
+
+        double PI = 3.14159265;
+
+        ArrayList<Double> lat_array = new ArrayList<Double>();
+        ArrayList<Double> long_array = new ArrayList<Double>();
+
+        for (String s : polygon_lat_long_pairs) {
+            lat_array.add(Double.parseDouble(s.split(",")[0]));
+            long_array.add(Double.parseDouble(s.split(",")[1]));
+        }
+
+        int i;
+        double angle = 0;
+        double point1_lat;
+        double point1_long;
+        double point2_lat;
+        double point2_long;
+        int n = lat_array.size();
+
+        for (i = 0; i < n; i++) {
+            point1_lat = lat_array.get(i) - latitude;
+            point1_long = long_array.get(i) - longitude;
+            point2_lat = lat_array.get((i + 1) % n) - latitude;
+            // you should have paid more attention in high school geometry.
+            point2_long = long_array.get((i + 1) % n) - longitude;
+            angle += Angle2D(point1_lat, point1_long, point2_lat, point2_long);
+        }
+
+        if (Math.abs(angle) < PI) {
+            return false;
+        } else {
+            return true;
+        }
+    }
+
+    public static double Angle2D(double y1, double x1, double y2, double x2) {
+
+        double PI = 3.14159265;
+        double TWOPI = 2 * PI;
+
+        double dtheta, theta1, theta2;
+
+        theta1 = Math.atan2(y1, x1);
+        theta2 = Math.atan2(y2, x2);
+        dtheta = theta2 - theta1;
+        while (dtheta > PI) {
+            dtheta -= TWOPI;
+        }
+        while (dtheta < -PI) {
+            dtheta += TWOPI;
+        }
+
+        return (dtheta);
+    }
+
+    public static boolean is_valid_gps_coordinate(double latitude, double longitude) {
```

Comment:
please use camel case method names. is there a special reason why this should be static? add JavaDoc

Reply:
Renamed JavaDoc added, don't know about static, my first JAVA program.

Reply:
`static` is usally used if the method is totally independent of the class itself (see the `Util` class I linked below). If you declare something as static therw ill be only one instance of that (which can be called from other classes without instantiating a object of that class before.

Reply:
Renamed , changed to private and removed static